### ISSUE TITLE
clang-tidy check bugprone-copy-constructor-init gives no warning with the example in the docs

### ISSUE THREAD

**Issue by sean-mcmanus** [llvm/llvm-project#55572]
With clang-tidy 14.0.3 (on Windows 11).

Run clang-tidy with the example in the docs at https://releases.llvm.org/14.0.0/tools/clang/tools/extra/docs/clang-tidy/checks/bugprone-copy-constructor-init.html

Specifically,
```cpp
class Copyable {
public:
  Copyable() = default;
  Copyable(const Copyable &) = default;
};
class X2 : public Copyable {
  X2(const X2 &other) {} // Copyable(other) is missing
};
```
with
```shell
clang-tidy.exe --config="{Checks: 'bugprone-copy-constructor-init'}" <path>/test.cpp -- -std=c++17
```

Bug: No warnings are reported. If I'm doing something wrong, maybe the docs could explain better what the requirements are.

**Comment by PiotrZSL** [comment]
Issue is not reported here, because check detect that base class has no members, and therefor it doesn't make any difference what constructor will be called. Only thing that should be done here, is documentation should be updated.

**Comment by PiotrZSL** [comment]
https://reviews.llvm.org/D144594
